frontend/src/components/OmakaseModal.jsx

```diff
@@ -0,0 +1,209 @@
+import { useState } from "react";
+import {
+  Modal,
+  Spinner,
+  FormField,
+  Select,
+  Input,
+  Checkbox,
+  ExpandableSection,
+  SpaceBetween,
+  Button,
+  Box,
+} from "@cloudscape-design/components";
+import useStore from "../stores/useStore";
+import useOmakaseStore from "../stores/useOmakaseStore";
+import { createInitialFormData, executeExport } from "../utils/omakase";
+
+export default function OmakaseModal({ editTimeranges, flows }) {
+  const [formData, setFormData] = useState(createInitialFormData(flows));
+  const [isSpinnerVisible, setIsSpinnerVisible] = useState(false);
+  const [showAdvancedContent, setShowAdvancedContent] = useState(false);
+  const addAlertItem = useStore((state) => state.addAlertItem);
+  const delAlertItem = useStore((state) => state.delAlertItem);
+
+  const { omakaseModalVisible, setOmakaseModalVisible } = useOmakaseStore(
+    (state) => state
+  );
+
+  const operations = ["Segment Concatenation", "Flow Creation"];
+  const formats = ["TS", "MP4"];
+  const isExportButtonDisabled =
+    formData.operation === "Flow Creation" && formData.label === "";
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
```

Comment:
I didn't think this was required from a default Cloudscape Button since is it not classed as a "submit button"?